server/src/server.rs

```diff
@@ -420,12 +418,8 @@ mod tests {
                 ..Default::default()
             };
             let got = server.create_database(name, rules).await.unwrap_err();
-            if let Error::InvalidDatabaseName { db_name: got } = got {
-                if got != name {
-                    panic!(format!("expected name {} got {}", name, got));
-                }
-            } else {
-                panic!("unexpected error");
+            if !matches!(got, Error::InvalidDatabaseName { source: _s }) {
```

Comment:
This could be a bit shorter as:

```suggestion
            if !matches!(got, Error::InvalidDatabaseName { .. }) {
```

since we're ignoring all the fields.